### DISCUSSION TITLE
Problem completing integration installation. (Solved).

### DISCUSSION THREAD

**Discussion by Billscot** [CJNE/ha-myenergi#572] [General]
Folks
I've got HA running on a VM, in Proxmox.
I've installed CJNE's fine integration (thank you for that) using the 'Manual Installation' method.
However, when I attempt to add the integration, having entered the hub's S/N and API key (and seemingly had them accepted) I'm receiving a 'Failed to set up' message. Please see photo.
If it helps, I've already got a good instance of HA with the ME integration, working successfully on a RPi. Before I start the Proxmox VM, I remove the ethernet cable from the RPi, in order to avoid any conflicts.
For info, I've already successfully installed the GivEnergy add-on to the HA on Proxmox, which is working well with Mosquitto MQTT broker, which I'm running in a LNXC on the same Proxmox server.
I'm already past the limits of my ability here, so please be gentle!

Any thoughts?

Thanks, in advance.
![image](https://github.com/user-attachments/assets/489f37f5-87b3-42a6-befc-b3e4e8695b58)

**Comment by G6EJD** [comment]
What are your logs reporting?

**Reply by Billscot** [reply]
Hi. The only logs I can see, are in the HA logbook, but there's nothing in there about this integration attempt.
Are the 'logs' which the message refers to, kept somewhere else?

**Comment by Billscot** [comment]
@G6EJD
Thanks - you said enough to put me on the right track...
I used Studio Code Server to inspect the 'home-assistant.log' and found this line:
'ModuleNotFoundError: Platform myenergi.switch not found'
...which led me to the custom_components directory, where I found the switch.py file outside the myenergi directory, instead of within it.
Moved the .py file to the correct place, removed the faulty myenergi integration, restarted HA, and added the integration with no issues.

Thanks for your input.

**Comment by G6EJD** [comment]
Good to hear it’s working now
